wallet/wallet.go

```diff
@@ -2464,6 +2464,73 @@ func (w *Wallet) GetTransactions(startBlock, endBlock *BlockIdentifier,
 	return &res, err
 }
 
+// GetTransactionResult returns a transaction in the UnminedTransaction
+// field if it is unmined, or in the MinedTransaction field as a Block
+// structure for a mined transaction.
+type GetTransactionResult struct {
+	MinedTransaction   Block
+	UnminedTransaction TransactionSummary
+}
+
+// GetTransaction returns data of any transaction given its id. A mined
+// transaction is returned in a Block structure which records properties about
+// the block.
+func (w *Wallet) GetTransaction(txHash *chainhash.Hash) (*GetTransactionResult,
+	error) {
+
+	var (
+		bestHeight int32
+		blockHash  *chainhash.Hash
+		summary    TransactionSummary
+	)
+	err := walletdb.View(w.db, func(dbtx walletdb.ReadTx) error {
+		txmgrNs := dbtx.ReadBucket(wtxmgrNamespaceKey)
+
+		txDetail, err := w.TxStore.TxDetails(txmgrNs, txHash)
+		if err != nil {
+			return err
+		}
+
+		// If the transaction was not found we can exit early.
+		if txDetail == nil {
+			return nil
+		}
+
+		// Otherwise, we create a summary of the transaction details.
+		summary = makeTxSummary(dbtx, w, txDetail)
+
+		// If it is a confirmed transaction we set the corresponding
+		// block height and hash.
+		if txDetail.Block.Height != -1 {
+			bestHeight = txDetail.Block.Height
+			blockHash = &txDetail.Block.Hash
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var res GetTransactionResult
```

Comment:
Have this be the only variable outside of the `walletdb.View` call and move the `switch` below inside the callback? That way it's more clear what the result will be if `txDetail` is nil.

Reply:
Great. I'll do that and return an error if the transaction isn't found.

Reply:
Done. Now I return an error if the transaction isn't found instead.